src/ds/ds6/ds6-active.cpp

```diff
@@ -0,0 +1,82 @@
+// License: Apache 2.0. See LICENSE file in root directory.
```

Comment:
As discussed 99-100% of this code is common.
We need to find a method for reuse.

Reply:
done